client/src/modules/reports/generate/inventory_report/inventory_report.html

```diff
@@ -0,0 +1,93 @@
+<bh-report-preview
+  ng-if="ReportConfigCtrl.previewGenerated"
+  source-document="ReportConfigCtrl.previewResult"
+  on-clear-callback="ReportConfigCtrl.clearPreview()"
+  on-save-callback="ReportConfigCtrl.requestSaveAs()">
+</bh-report-preview>
+
+<div ng-show="!ReportConfigCtrl.previewGenerated">
+  <div class="row">
+    <div class="col-md-12">
+      <h3 class="text-capitalize" translate>REPORT.STOCK.TITLE</h3>
+      <p class="text-info" translate>REPORT.STOCK.DESCRIPTION</p>
+    </div>
+  </div>
+
+  <div class="row" style="margin-top : 10px">
+    <div class="col-md-6">
+      <div class="panel panel-default">
+        <div class="panel-heading">
+          <span translate>REPORT.UTIL.OPTIONS</span>
+        </div>
+
+        <div class="panel-body">
+
+        <form name="ConfigForm" bh-submit="ReportConfigCtrl.preview(ConfigForm)" novalidate autocomplete="off">
+
+          <!-- choose date until  -->
+          <bh-date-editor
+            label="FORM.LABELS.UNTIL_PERIOD"
+            date-value="ReportConfigCtrl.dateTo">
+          </bh-date-editor>
+
+          <!-- delay -->
+          <div class="form-group col-xs-12 col-sm-6">
+            <label class="control-label" translate>FORM.LABELS.DELAY</label>
+              <div class="input-group">
+              <input class="form-control" type="number" step="1" name="delay" ng-model="ReportConfigCtrl.delay">
+              <span class="input-group-addon" translate>FORM.LABELS.MONTH</span>
+            </div>
+          </div>
+
+          <!-- purchase interval -->
+          <div class="form-group col-xs-12 col-sm-6">
+            <label class="control-label" translate>FORM.LABELS.PURCHASE_INTERVAL</label>
+            <div class="input-group">
+              <input class="form-control" type="number" step="1" name="purchase_interval" ng-model="ReportConfigCtrl.purchaseInterval">
+              <span class="input-group-addon" translate>FORM.LABELS.MONTH</span>
+            </div>
+          </div>
+
+          <!-- depot options -->
+          <div class="checkbox">
+            <label>
+              <input type="checkbox" ng-model="ReportConfigCtrl.chooseOneDepot" ng-true-value="1" ng-false-value="0">
+              <span translate>REPORT.STOCK.ONE_DEPOT</span>
+            </label>
+          </div>
+
+          <!-- select depot -->
+          <div ng-if="!!ReportConfigCtrl.chooseOneDepot">
```

Comment:
I think both the `<bh-depot-select>` and the `<bh-inventory-select>` should be required if they are shown.

Without them required, this bug exists:
 1. Navigate to this page.
 2. Check both the checkboxes to show the `select`s.
 3. Click Submit 
 4. Observe that the button loads forever.  The console reveals:
> TypeError: Cannot read property 'uuid' of undefined